js/woo.js

```diff
@@ -0,0 +1,1465 @@
+        }
+        const transactionType = this.safeString (params, 'transaction_type');
+        if (transactionType !== undefined) {
+            request['type'] = transactionType;
+        }
+        params = this.omit (params, 'type'); // conflict! we need to change 'type' param for spot/futures into 'exchangeType'
+        const [ marketType, query ] = this.handleMarketTypeAndParams ('fetchTransactions', undefined, params);
+        const method = this.getSupportedMapping (marketType, {
+            'spot': 'v1PrivateGetAssetHistory',
+        });
+        const response = await this[method] (this.extend (request, query));
+        // {
+        //     rows: [
+        //       {
+        //         id: '22010508193900165',
+        //         token: 'TRON_USDT',
+        //         extra: '',
+        //         amount: '13.75848500',
+        //         status: 'COMPLETED',
+        //         account: null,
+        //         description: null,
+        //         user_id: '42222',
+        //         application_id: '6ad2b303-f354-45c0-8105-9f5f19d0e335',
+        //         external_id: '220105081900134',
+        //         target_address: 'TXnyFSnAYad3YCaqtwMw9jvXKkeU39NLnK',
+        //         source_address: 'TYDzsYUEpvnYmQk4zGP9sWWcTEd2MiAtW6',
+        //         type: 'BALANCE',
+        //         token_side: 'DEPOSIT',
+        //         tx_id: '35b0004022f6b3ad07f39a0b7af199f6b258c2c3e2c7cdc93c67efa74fd625ee',
+        //         fee_token: '',
+        //         fee_amount: '0.00000000',
+        //         created_time: '1641370779.442',
+        //         updated_time: '1641370779.465',
+        //         is_new_target_address: null,
+        //         confirmed_number: '29',
+        //         confirming_threshold: '27',
+        //         audit_tag: '1',
+        //         audit_result: '0',
+        //         balance_token: null, // TODO -write to support, that this seems broken
+        //         network_name: null // TODO -write to support, that this seems broken
+        //       }
+        //     ],
+        //     meta: { total: '1', records_per_page: '25', current_page: '1' },
+        //     success: true
+        // }
+        const transactions = this.safeValue (response, 'rows', {});
+        return this.parseTransactions (transactions, currency, since, limit);
+    }
+
+    parseTransaction (transaction, currency = undefined) {
+        // example in fetchTransactions
+        const networkizedCode = this.safeString (transaction, 'token');
+        if (currency === undefined) {
+            currency = this.getCurrencyByNetworkizedCode (networkizedCode);
+        }
+        const code = currency['code'];
+        let movementDirection = this.safeStringLower (transaction, 'token_side');
+        if (movementDirection === 'withdraw') {
+            movementDirection = 'withdrawal';
+        }
+        let fee_token = this.safeString (transaction, 'fee_token', '');
```

Comment:
We never pass the exchange specific currency ids (`fee_token` in this case) to the user "as is". We always convert them from exchange-specific currency ids, to CCXT unified currency codes. That is done with the following methods:

```JavaScript
const feeCurrencyId = this.safeString (transaction, 'fee_token');
const feeCurrencyCode = this.safeCurrencyCode (feeCurrencyId);
```

See how that is done across the other exchanges.